packages/model-generator/src/create_graphql_models_generator.ts

```diff
@@ -123,12 +122,12 @@ const getModelSchema = async (
 
 const getModelSchemaFromS3Uri = async (
   modelSchemaS3Uri: string,
-  awsClientProvider: AWSClientProvider<{
+  awsClientProvider?: AWSClientProvider<{
     getS3Client: S3Client;
   }>
 ): Promise<string> => {
   const schemaFetcher = new S3StringObjectFetcher(
-    awsClientProvider.getS3Client()
+    awsClientProvider?.getS3Client() ?? new S3Client()
```

Comment:
rather than plumbing this as optional all the way down here, we should instantiate defaults as early in the call stack as possible and have a required arg the rest of the way down. Applies to plumbing other clients as well.